src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/MultipleField.jsx

```diff
@@ -22,19 +22,31 @@ const styles = theme => {
     };
 };
 
+const _mapDataForOnChange = dataToMap => {
+    return dataToMap.filter(item => item.data !== undefined).map(item => item.data);
+};
+
+const reducer = (state, action) => {
+    switch (action.type) {
+        case 'set':
+            return action.data;
+        case 'unMount':
+            return action.onChange(_mapDataForOnChange(state), undefined);
+        default:
+            throw new Error();
+    }
+};
+
 export const MultipleFieldCmp = ({classes, inputContext, field, onChange, formik: {values}}) => {
     const {t} = useTranslation();
     const [isInit, setInit] = useState(false);
-    const [data, setData] = useState(values[field.name] ? new Array(values[field.name].length) : []);
 
+    const [data, dispatch] = useReducer(reducer, values[field.name] ? new Array(values[field.name].length) : []);
+    const setData = value => dispatch({type: 'set', data: value});
     useEffect(() => {
-        return () => onChange(_mapDataForOnChange(data), undefined);
+        return () => dispatch({type: 'unMount', onChange});
     }, []);
```

Comment:
at least we can add parameter to the array of useEffect like: value, to optimize number of calls

Reply:
no we can't as the state is not reachable by the closure.

Reply:
I don't see how it's not reachable to do [field.value] ...